src/Backends/InterfaceSettings.vala

```diff
@@ -63,8 +86,43 @@ public class SettingsDaemon.Backends.InterfaceSettings : GLib.Object {
         accounts_service.cursor_size = interface_settings.get_int (CURSOR_SIZE);
         accounts_service.locate_pointer = interface_settings.get_boolean (LOCATE_POINTER);
         accounts_service.text_scaling_factor = interface_settings.get_double (TEXT_SCALING_FACTOR);
+
+        accounts_service.picture_options = background_settings.get_enum (PICTURE_OPTIONS);
+        accounts_service.primary_color = background_settings.get_string (PRIMARY_COLOR);
+
         accounts_service.document_font_name = interface_settings.get_string (DOCUMENT_FONT_NAME);
         accounts_service.font_name = interface_settings.get_string (FONT_NAME);
         accounts_service.monospace_font_name = interface_settings.get_string (MONOSPACE_FONT_NAME);
     }
+
+    private void sync_background_to_greeter () {
+        var source = File.new_for_uri (background_settings.get_string (PICTURE_URI));
+
+        if (!source.query_exists ()) {
+            debug ("Wallpaper path is invalid");
+            display_manager_accounts_service.background_file = "";
+            return;
+        }
+
+        var wallpaper_name = "wallpaper";
+
+        var greeter_data_dir = Environment.get_variable ("XDG_GREETER_DATA_DIR") ?? Path.build_filename ("/var/lib/lightdm-data", Environment.get_user_name ());
+        var folder = File.new_for_path (greeter_data_dir);
+
+        try {
+            if (!folder.query_exists ()) {
+                folder.make_directory_with_parents ();
+            }
+
+            source.copy (folder.get_child (wallpaper_name), FileCopyFlags.OVERWRITE | FileCopyFlags.ALL_METADATA);
+            // Ensure wallpaper is readable by greeter user (owner rw, others r)
+            FileUtils.chmod (folder.get_child (wallpaper_name).get_path (), 0604);
+
+            display_manager_accounts_service.background_file = folder.get_child (wallpaper_name).get_path ();
+        } catch (Error e) {
+            warning (e.message);
+            display_manager_accounts_service.background_file = "";
+            return;
+        }
```

Comment:
i believe we want to deal with the case of `GLib.IOError.IS_DIRECTORY` as a migration path, since we are moving from `lightdm-data/$USER/wallpaper/$WALLPAPER_FILE` to `lightdm-data/$USER/wallpaper`.